### DISCUSSION TITLE
FR: Fuzzy find for file picker and select

### DISCUSSION THREAD

**Discussion by chlunde** [charmbracelet/huh#557] [Ideas]
**Is your feature request related to a problem? Please describe.**
Scrolling through long file/option lists is slow, for example when there are more than 20 options. I could not find a easy way to get fuzzy finding with `huh`.

**Describe the solution you'd like**
Fuzzy finding, similar to `fzf`, would allow reducing the number of options quickly.

For API, `huh.NewSelect().WithFuzzyFind(true)`, or huh.NewSelect().WithFuzzy(huh.DefaultFuzzyFinder)` could be nice

**Describe alternatives you've considered**
It might be possible by using two existing fields: a dynamic select box, and an input box for filtering the items in the select box. If this is the best solution, an example would be nice!

Related: https://github.com/charmbracelet/huh/issues/262

**Comment by bashbunni** [comment]
Hey @chlunde, thank you so much for wanting to make this project better! I've converted this to a discussion so we can talk through it and keep our backlog clear. If it does end up being something we want to add, we'll promote it back to an issue when we are ready to work on it.

If we commit to this feature, that's a feature we will need to maintain for the lifespan of this project. So, since "Yes is forever", we are carefully considering new features.

This issue has gotten a lot of attention from the community, so clearly it's a feature that other users agree would be valuable. I do think being able to search for a specific file or folder name makes sense for a `filepicker`. I don't recall if we have this feature already in bubbles, but this should probably exist in the `filepicker` bubble. I'll start a discussion in that repo too.

**Reply by bashbunni** [reply]
https://github.com/charmbracelet/bubbles/discussions/747
